testing/scenario_app/android/app/src/main/java/dev/flutter/scenarios/ExternalTextureFlutterActivity.java

```diff
@@ -92,7 +94,9 @@ public void waitUntilFlutterRendered() {
     super.waitUntilFlutterRendered();
 
     try {
-      firstFrameLatch.await(10, java.util.concurrent.TimeUnit.SECONDS);
+      if (!firstFrameLatch.await(10, java.util.concurrent.TimeUnit.SECONDS)) {
+        throw new RuntimeException("Timeout waiting for firstFrameLatch to signal");
```

Comment:
There is a class for timeouts: https://docs.oracle.com/javase/8/docs/api/java/util/concurrent/TimeoutException.html

Reply:
It's a checked exception though, I don't think we actually want our callers to catch anything (it's not actionable).